### PR TITLE
CASMHMS-4815 / CASMHMS-5803 / CASMHMS-5877 - Action retention, log file cleanup, code cleanup

### PR DESCRIPTION
**Pull request by mbuchmann-hpe**
### Summary and Scope

Auto removal of actions by age:
> FAS can now remove actions which are >x days old.  The age is variable and can be defined with an environment variable or on the command line.  A value of 0 (zero) will not delete any actions.  The default is 0.

Fixed error / warning messages:
> FAS will not query the database for null ids.  This removes error messages from logs.
> Correct warning messages "state is nil".

Code cleanup:
> Cleaned up indenting and spacing.
> Cleaned up delete snapshot code.

Removed database dump and load from this release.

DOES THIS CHANGE INVOLVE ANY SCHEME CHANGES?  N

REMINDER: HAVE YOU INCREMENTED VERSION NUMBERS? Y

REMINDER 2: HAVE YOU UPDATED THE COPYRIGHT PER hpe GUIDELINES: (C) Copyright 2014-2021 Hewlett Packard Enterprise Development LP?  Y

### Issues and Related PRs

CASMHMS-4815 - Create tuneable retention policy for actions
CASMHMS-5803 - Remove error message for key not found on a blank image
CASMHMS-5877 - Logs contain many warning "State is nil"

### Testing

Tested locally on a simulated environment.
Tested on Hela.

Ran CT, Unit, and Integration tests

### Risks and Mitigations

No FAS functionality was changed.

### CODE REVIEW

File: internal/domain/service_DB.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import (
+	"net/http"
+
+	"github.com/Cray-HPE/hms-firmware-action/internal/model"
+	"github.com/Cray-HPE/hms-firmware-action/internal/storage"
+	"github.com/sirupsen/logrus"
+)
+
+func DumpDB() (pb model.Passback) {
```
Comment (nieuwsma) [inline internal/domain/service_DB.go:35]: @rsjostrand-hpe had a good idea.  Lets move the dump/load stuff into a separate application in CMD directory. WE could spin it up in docker compose as necessary, but you could run it along side the FAS binary as needed.

**Review by nieuwsma — CHANGES_REQUESTED** [review]

File: internal/domain/service_DB.go (hunk 1 above)
Comment (mbuchmann-hpe) [reply internal/domain/service_DB.go]: Removed dump and load database from this release.

**Review by rsjostrand-hpe — APPROVED** [review]

File: cmd/hms-firmware-action/updateScheduler.go [hunk 2]
```diff
@@ -103,17 +103,22 @@ func controlLoop(domainGlobal *domain.DOMAIN_GLOBALS) {
...
-			pb := domain.DeleteSnapshot(snapshot.Name)
-			if pb.IsError {
-				mainLogger.Error(pb.Error.Detail)
-			}
+		// Check for expired snapshots and actions
+		// Do not have to do this every time through the loop
+		if checkExpiredItems <= 0 {
+			domain.DeleteExpiredSnapshots()
+			domain.DeleteExpiredActions(domainGlobal.DaysToKeepActions)
+			checkExpiredItems = 500
```
Comment (nieuwsma) [inline cmd/hms-firmware-action/updateScheduler.go:119]: why is this set to 500?

Comment (mbuchmann-hpe) [reply cmd/hms-firmware-action/updateScheduler.go:119]: I did not want to check for expired items every time through the loop, so I set a simple counter.   500 would be a little more then once an hour check.

**Review by nieuwsma — APPROVED** [review]

**Pull request merged**